test/system/settings_test.rb

```diff
@@ -46,6 +76,28 @@ class SettingsTest < ApplicationSystemTestCase
     assert_selector 'span[data-clipboard-target="iconSuccess"]', visible: true, count: 1 # text copied and icon changed to checkmark
   end
 
+  test "can clear data cache in self hosting settings" do
+    Rails.application.config.app_mode.stubs(:self_hosted?).returns(true)
+    open_settings_from_sidebar
+    click_link I18n.t("settings.settings_nav.self_hosting_label")
+
+    assert_selector "h3", text: I18n.t("settings.hostings.show.clear_cache")
+    assert_selector "p", text: I18n.t("settings.hostings.show.clear_cache_warning")
+
+    clear_cache_button = find_button I18n.t("settings.hostings.show.clear_cache")
+
+    clear_cache_button.click
+
+    within("#turbo-confirm") do
+      assert_text I18n.t("settings.hostings.show.confirm_clear_cache.title")
+      assert_text I18n.t("settings.hostings.show.confirm_clear_cache.body")
+      click_button I18n.t("settings.hostings.show.clear_cache")
+    end
+
+    assert_current_path settings_hosting_path
+    assert_text I18n.t("settings.hostings.clear_cache.cache_cleared")
+  end
```

Comment:
Could we remove these system tests and keep all our testing for this feature in controller/job tests?

We're trying to keep the system test suite as minimal and stable as possible for CI speed and reliability, and I think we're adding in a lot of dependence on the UI structure here, which may change in the future.  The most important tests I think we need for this feature is to validate that:

- Jobs actually delete the intended data
- Admins can't trigger these jobs ever, regardless of what the UI looks like (the UI guards should be purely visual aids)